### PR TITLE
Clean BuildFiles

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/42336).

In some cases, the dependencies had to be moved to the appropriate BuildFile, for example from a plugins directory to the library directory or vice versa.

This PR cleans unnecessary includes from CMSSW BuildFiles that were added in the last months.

#### PR validation:

CMSSW compiles.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

### CODE REVIEW

**Comment by srimanob** [comment]
@cmsbuild please test

File: L1Trigger/L1TZDC/BuildFile.xml
Comment (makortel) [inline L1Trigger/L1TZDC/BuildFile.xml:1]: Seems like this whole file should be removed.

File: CalibTracker/SiStripHitResolution/plugins/BuildFile.xml [hunk 1]
```diff
@@ -4,7 +4,6 @@
 <use name="CommonTools/TrackerMap"/>
 <use name="CommonTools/UtilAlgos"/>
 <use name="DataFormats/Common"/>
-<use name="DataFormats/Scalers"/> <!--Needed for typeinfo for LumiScalers for UBSAN builds-->
```
[the remark this answers is not in the document]
Comment (makortel) [reply CalibTracker/SiStripHitResolution/plugins/BuildFile.xml:7]: I got worried about the same. I think this PR should be tested against the UBSAN IB.

File: PerfTools/AllocMonitor/test/BuildFile.xml [hunk 2]
```diff
@@ -5,19 +5,16 @@
...
 
 <bin file="test_proxies.cc" name="testAllocMonitorPreloadTC">
   <use name="PerfTools/AllocMonitor"/>
-  <use name="PerfTools/AllocMonitorPreload"/>
   <use name="tcmalloc_minimal"/>
   <flags CXXFLAGS="-O0"/>
 </bin>
 <bin file="test_proxies.cc" name="testAllocMonitorPreloadJE">
   <use name="PerfTools/AllocMonitor"/>
-  <use name="PerfTools/AllocMonitorPreload"/>
```
[the remark this answers is not in the document]
Comment (guitargeek) [reply PerfTools/AllocMonitor/test/BuildFile.xml:20]: Alright! I'll add this back.

File: CalibTracker/SiStripHitResolution/plugins/BuildFile.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (guitargeek) [reply CalibTracker/SiStripHitResolution/plugins/BuildFile.xml:7]: Sorry, that was an oversight. I didn't mean to remove it. I'll bring that line back

File: L1Trigger/L1TZDC/BuildFile.xml
Comment (guitargeek) [reply L1Trigger/L1TZDC/BuildFile.xml:1]: Will do!

**Comment by guitargeek** [comment]
I have updated the PR following the review.

**Comment by guitargeek** [comment]
Sorry for the bot adding that unnecessary label. I was making and pushing the commit from a machine where the clock was off by a minute, I hope this is not a problem.

**Comment by smuzaffar** [comment]
please test

File: PerfTools/AllocMonitor/test/BuildFile.xml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <bin file="test_catch2_*.cc" name="testPerfToolsAllocMonitorCatch2">
   <use name="PerfTools/AllocMonitor"/>
+  <use name="PerfTools/AllocMonitorPreload"/>
```
Comment (makortel) [inline PerfTools/AllocMonitor/test/BuildFile.xml:3]: This test should be left without the `AllocMonitorPreload`
```suggestion
```

**Comment by emanueleusai** [comment]
please test

**Comment by AdrianoDee** [comment]
please test
(retry)

**Comment by AdrianoDee** [comment]
please abort

**Comment by guitargeek** [comment]
Hi, can anyone start the tests again here?

**Comment by smuzaffar** [comment]
please test

**Comment by makortel** [comment]
@cmsbuild, please test for CMSSW_13_3_UBSAN_X

Let's double-check UBSAN IB as well.

**Comment by makortel** [comment]
UBSAN built fine, good

**Comment by guitargeek** [comment]
Hi @makortel, if the UBSAN built fine, why are the tests failing? I'm confused about the status of this PR :)

**Comment by mmusich** [comment]
> I found 1 errors in the following unit tests:
> `---> test CondToolsLHCInfoNewPopConTest had ERROR`

This already fails in IBs, see https://github.com/cms-sw/cmssw/issues/42828

**Comment by guitargeek** [comment]
I see, thanks! Then it's just about collecting the signatures not I guess

**Comment by makortel** [comment]
> I see, thanks! Then it's just about collecting the signatures not I guess

Right, the PR looks good as far as I can tell. So everyone involved should just sign now.

**Comment by emanueleusai** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by aloeliger** [comment]
+l1

**Comment by srimanob** [comment]
+Upgrade

**Comment by perrotta** [comment]
+alca

**Comment by perrotta** [comment]
+db

**Comment by civanch** [comment]
+1

**Comment by simonepigazzini** [comment]
+xpog

**Comment by guitargeek** [comment]
Hi @smuzaffar, it looks like the bot had a problem in updating the `xpog` label, even though the correct person was signing:
https://github.com/cms-sw/cms-bot/blob/master/categories.py#L50

Can you maybe help out here? Thanks!

**Comment by smuzaffar** [comment]
ping bot

**Comment by antoniovilela** [comment]
+1

**Comment by antoniovilela** [comment]
merge

**Pull request merged**